addons/repair/models/account_move.py

```diff
@@ -9,7 +9,8 @@ class AccountMove(models.Model):
     repair_ids = fields.One2many('repair.order', 'invoice_id', readonly=True, copy=False)
 
     def unlink(self):
-        self.repair_ids.filtered(lambda repair: repair.state != 'cancel').state = '2binvoiced'
+        if self.sudo().repair_ids:
```

Comment:
Nice.
just a small details:

IMO, you can add the filtered in the check to avoid raise when the repair is canceled for no reason.
```python
repairs = self.sudo().repair_ids.filtered(lambda repair: repair.state != 'cancel')
if repairs:
   repairs.sudo(False).state = '2binvoiced'
```
(to test)

Reply:
Thanks @ryv-odoo for the review !
After testing, it seems to work correctly